subgraphs/aave-v2/src/mappings/utilFunctions.ts

```diff
@@ -0,0 +1,605 @@
+  const incentivesController = initIncentivesController(market);
+  // From the incentives controller contract, get pull the 'assets' values from the market aToken, sToken, and vToken
+  const incentivesControllerContract = IncentivesControllerContract.bind(incentivesController);
+  if (!incentivesControllerContract.try_assets(Address.fromString(market.outputToken)).reverted) {
+    log.info('REWARD EMISSIONS ' + market.outputToken + ' ' + incentivesControllerContract.assets(Address.fromString(market.outputToken)).value0.toString() + ' ' + incentivesControllerContract.assets(Address.fromString(market.outputToken)).value1.toString() + ' ' + incentivesControllerContract.assets(Address.fromString(market.outputToken)).value2.toString(), []);
+    const assetDataSupply = incentivesControllerContract.try_assets(Address.fromString(market.outputToken)).value.value0;
+    const assetDataBorrowStable = incentivesControllerContract.try_assets(Address.fromString(market.sToken)).value.value0;
+    const assetDataBorrowVariable = incentivesControllerContract.try_assets(Address.fromString(market.sToken)).value.value0;
+    // Get the emissions per day for the aToken rewards for deposits
+    rewardEmissions[0] = emissionsPerDay(assetDataSupply);
+    // Get the emissions per second for both the sToken and vToken rewards, average them and get the daily emissions for borrows
+    rewardEmissions[1] = emissionsPerDay((assetDataBorrowStable.plus(assetDataBorrowVariable)).div(BIGINT_TWO));
+  } else {
+    log.info('COULD NOT GET REWARD EMISSIONS FROM INC CONT ON MARKET: ' + market.id, []);
+    // The rewardEmissions array returned is already defaulted to zero vals
+  }
+  return rewardEmissions;
+}
+
+export function getOutputTokenSupply (event: ethereum.Event): void {
+  const outputTokenAddr = event.address;
+  const aTokenInstance = AToken.bind(outputTokenAddr);
+  const tryTokenSupply = aTokenInstance.try_totalSupply();
+  if (!tryTokenSupply.reverted) {
+    log.info('OUTPUT TOKEN NOT REVERTED ' + outputTokenAddr.toHexString() + ' ' + tryTokenSupply.value.toString(), []);
+    const market = Market.load(aTokenInstance.UNDERLYING_ASSET_ADDRESS().toHexString());
+    if (!market) return;
+    market.outputTokenSupply = tryTokenSupply.value;
+    market.save();
+    // This funcion only gets called from mint/burn events which only happen following deposits and withdraws which update the market snapshot
+    // Calling the getMarketDailySnapshot function seems to cause some sort of overflow error, so only the outputTokenSupply field needs to be set
+    const snapId = getMarketDailyId(event, market);
+    const snap = MarketDailySnapshot.load(snapId);
+    if (!snap) return;
+    snap.outputTokenSupply = market.outputTokenSupply;
+    snap.save();
+  } else {
+    log.info('OUTPUT TOKEN SUPPLY CALL REVERTED FOR TOKEN ' + outputTokenAddr.toHexString(), []);
+  }
+}
+
+export function initToken (assetAddr: Address): Token {
```

Comment:
why not use `getOrCreateToken` in `common\getters.ts`?

Reply:
I originally wrote this function before the common scripts were added. I just removed the initToken function and changed all references to the 'getOrCreateToken' function. I added the appropriate ABIs and adjusted the schema to remove errors from the functions in 'token.ts' file that are called when the 'getOrCreateToken' function is executed.

Reply:
I originally wrote this function before the common scripts were added. I just removed the initToken function and changed all references to the 'getOrCreateToken' function. I added the appropriate ABIs and adjusted the schema to remove errors from the functions in 'token.ts' file that are called when the 'getOrCreateToken' function is executed.